sdk/monitor/Azure.Monitor.Ingestion/tests/IngestionClientLiveTest.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using Azure.Core;
+using Azure.Core.TestFramework;
+using Azure.Identity;
+using NUnit.Framework;
+using Azure.Monitor.Query;
+using Azure.Monitor.Query.Models;
+using System.Threading.Tasks;
+using System.Linq;
+
+namespace Azure.Monitor.Ingestion.Tests
+{
+    public class IngestionClientLiveTest : RecordedTestBase<IngestionClientTestEnvironment>
+    {
+        public IngestionClientLiveTest(bool isAsync) : base(isAsync)
+        {
+        }
+
+        /* please refer to https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/template/Azure.Template/tests/TemplateClientLiveTests.cs to write tests. */
+
+        private LogsIngestionClient CreateClient()
+        {
+            return new LogsIngestionClient(new Uri(TestEnvironment.DCREndpoint), TestEnvironment.ClientSecretCredential);
+        }
+
+        [Test]
+        public async Task ValidInputFromObjectAsJson()
+        {
+            LogsIngestionClient client = CreateClient();
+            var currentTime = Recording.UtcNow;
+
+            BinaryData data = BinaryData.FromObjectAsJson(
+                // Use an anonymous type to create the payload
+                new[] {
+                    new
+                    {
+                        Time = currentTime,
+                        Computer = "Computer1",
+                        AdditionalContext = new
+                        {
+                            InstanceName = "user1",
+                            TimeZone = "Pacific Time",
+                            Level = 4,
+                            CounterName = "AppMetric1",
+                            CounterValue = 15.3
+                        }
+                    },
+                    new
+                    {
+                        Time = currentTime,
+                        Computer = "Computer2",
+                        AdditionalContext = new
+                        {
+                            InstanceName = "user2",
+                            TimeZone = "Central Time",
+                            Level = 3,
+                            CounterName = "AppMetric1",
+                            CounterValue = 23.5
+                        }
+                    },
+                });
+
+            // Make the request
+            Response response = await client.UploadAsync(TestEnvironment.DCRImmutableId, "Custom-MyTableRawData", RequestContent.Create(data)).ConfigureAwait(false); //takes StreamName not tablename
+
+            // Check the response
+            Assert.AreEqual(204, response.Status);
+
+            LogsQueryClient logsQueryClient = new(TestEnvironment.ClientSecretCredential);
+            var batch = new LogsBatchQuery();
+
+            string countQueryId = batch.AddWorkspaceQuery(
+                TestEnvironment.WorkspaceId,
+                "MyTable_CL | count;",
```

Comment:
Pull the name `MyTable_CL` into your test environment since it'll come from the ARM template

Reply:
made a note for later release